src/libsyntax/visit.rs

```diff
@@ -465,8 +465,8 @@ pub fn walk_pat<'a, V: Visitor<'a>>(visitor: &mut V, pattern: &'a Pat) {
         }
         PatKind::Lit(ref expression) => visitor.visit_expr(expression),
         PatKind::Range(ref lower_bound, ref upper_bound, _) => {
-            visitor.visit_expr(lower_bound);
-            visitor.visit_expr(upper_bound);
+            walk_list!(visitor, visit_expr, lower_bound);
```

Comment:
Use `if let Some`

Reply:
Like in `intravisit`, we use `walk_list!` to visit `Option<...>`s as well for the most part -- imo this shouldn't change.